core/services/relay/evm/functions/logpoller_wrapper.go

```diff
@@ -57,22 +56,39 @@ type detectedEvents struct {
 	detectedEventsOrdered []detectedEvent
 }
 
-type coordinator struct {
-	address        common.Address
-	typeAndVersion string
+type Coordinator interface {
+	Address() common.Address
+	RegisterFilters() error
+	OracleRequestLogTopic() (common.Hash, error)
+	OracleResponseLogTopic() (common.Hash, error)
+	LogsToRequests(requestLogs []logpoller.Log) ([]evmRelayTypes.OracleRequest, error)
+	LogsToResponses(responseLogs []logpoller.Log) ([]evmRelayTypes.OracleResponse, error)
 }
 
 const FUNCTIONS_COORDINATOR_VERSION_1_SUBSTRING = "Functions Coordinator v1"
 const FUNCTIONS_COORDINATOR_VERSION_2_SUBSTRING = "Functions Coordinator v2"
 
-type Functions_Coordinator interface {
-	*functions_coordinator_1_1_0.FunctionsCoordinator110 | *functions_coordinator.FunctionsCoordinator
-}
-
 const logPollerCacheDurationSecDefault = 300
 const pastBlocksToPollDefault = 50
 const maxLogsToProcess = 1000
 
+var (
+	uint32Type  abi.Type
+	uint40Type  abi.Type
+	uint64Type  abi.Type
+	uint72Type  abi.Type
+	uint96Type  abi.Type
+	addressType abi.Type
+	bytes32Type abi.Type
+)
+
+func init() {
+	err := initAbiTypes()
+	if err != nil {
+		panic(err)
```

Comment:
Why are we introducing a panic? This does not seem like the best practice.

Reply:
Actually, I am not quite sure I understand why this function is here at all...

Reply:
Yeah I don't love the panic either, my initial reasons were:
The init() function can not return an error, so if anything fails this is the only way. The benefits of initialising this here is that it will run after variables are declare and before anything else is executed (this is specially important because these are pkg variables that can be reference from outside). Another pro: It will only run once compare to re-declaring the types on each request.

>We can probably run initAbiTypes() inside NewLogPollerWrapper() and also not panic.

I thought about this but, wouldn't we leave the door open to someone referencing these pkg variables before they are initialised? 🤔 

I think there is an alternative that could work too without panic and is defining a structure like:
```
type validAbiTypes struct{
    uint32Type abi.Type,
    ...
}
 ``` 
 that contain all the valid types as part of logpuller_wrapper, this could be initialised in the NewLogPollerWrapper and injected down to the V1 and V2 implementations of the Coordinator. 

will explore these two last options and have something for today